src/main/java/net/imagej/convert/NumericTypeToBigIntegerConverter.java

```diff
@@ -0,0 +1,71 @@
+/*-
+ * #%L
+ * ImageJ2 software for multidimensional image processing and analysis.
+ * %%
+ * Copyright (C) 2009 - 2022 ImageJ2 developers.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package net.imagej.convert;
+
+import java.math.BigInteger;
+
+import org.scijava.Priority;
+import org.scijava.convert.AbstractConverter;
+import org.scijava.convert.Converter;
+import org.scijava.plugin.Plugin;
+
+import net.imglib2.type.numeric.IntegerType;
+
+/**
+ * {@link Converter} used to convert {@link IntegerType}s to {@link BigInteger}s
+ *
+ * @author Gabriel Selzer
+ */
+@SuppressWarnings("rawtypes")
+@Plugin(type = Converter.class, priority = Priority.LOW)
+public class NumericTypeToBigIntegerConverter extends
+	AbstractConverter<IntegerType, BigInteger>
+{
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public <T> T convert(final Object src, final Class<T> dest) {
+		if (!dest.equals(BigInteger.class)) //
+			throw new IllegalArgumentException(dest + "must be BigInteger.class!");
```

Comment:
It doesn't have to be `BigInteger.class`, but rather only a supertype. I.e.: `dest.isAssignableFrom(BigInteger.class)`.